client/src/components/Auth/AuthProvider.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react"
+import axios from "axios"
+import jwt from "jwt-decode"
+import { useNavigate } from "react-router-dom"
+
+const AuthContext = React.createContext(null)
+
+const AuthProvider = ({ children }) => {
+  const [auth, setAuth] = useState(null)
+  const navigate = useNavigate()
+
+  const login = async (email, password, callback) => {
+    console.log("[Login]")
+    try {
+      const authResponse = await axios.post(
+        "http://localhost:8000/auth/login",
+        { email: email, password: password },
+        { "content-type": "application/json" }
+      )
+      const decoded = jwt(authResponse.data.token)
+      setAuth({ token: authResponse.data.token, user: decoded.user })
+      callback()
+    } catch (err) {
+      console.log(`Login error ${err}`)
+      // Assignment: what should we do if this fails?
+      //navigate back to login page
+      alert("password or email input error")
+      navigate("/login")
+    }
+  }
+
+  const register = (email, password, callback) => {
+    // Assignment: how do we register someone?
+    console.log("[Register]")
+    try {
+      const authResponse = axios.post(
+        "http://localhost:8000/auth/register",
+        { email: email, password: password },
+        { "content-type": "application/json" }
+      )
+      const decoded = jwt(authResponse.data.token)
+      setAuth({ token: localStorage.getItem("token"), user: decoded.user })
```

Comment:
Point the second: You're not actually using the token you think you're getting back. Instead you're getting a token from local storage. What happens if it isn't set?